### ISSUE TITLE
Running a test on a Windows executor from a Linux host is not possible

### ISSUE THREAD

**Issue by tjgq** [bazelbuild/bazel#19209]
It's currently not possible to run a `*_test` rule on a remote Windows executor when Bazel is running on a Linux host.

- Every test rule has a dependency on [`@bazel_tools//tools/test:test_wrapper`](https://cs.opensource.google/bazel/bazel/+/master:src/main/java/com/google/devtools/build/lib/analysis/BaseRuleClasses.java;l=217;drc=0e899c5d9289771908648ce4362c45c2171def8d) and [`@bazel_tools//tools/test:xml_writer`](https://cs.opensource.google/bazel/bazel/+/master:src/main/java/com/google/devtools/build/lib/analysis/BaseRuleClasses.java;l=222;drc=0e899c5d9289771908648ce4362c45c2171def8d), which point to bundled precompiled [`tw.exe`](https://cs.opensource.google/bazel/bazel/+/master:tools/test/BUILD.tools;l=53;drc=ee3637c19c131cf295eaa3fb59ea03f6e9a1f60e) and [`xml.exe`](https://cs.opensource.google/bazel/bazel/+/master:tools/test/BUILD.tools;l=64;drc=ee3637c19c131cf295eaa3fb59ea03f6e9a1f60e) files if Bazel was built for Windows, and are missing otherwise (causing the test rule to fail during loading).
- In addition, some test rules (including `sh_test`) have a dependency on [`@bazel_tools//tools/launcher:launcher`](https://cs.opensource.google/bazel/bazel/+/master:src/main/java/com/google/devtools/build/lib/bazel/rules/sh/BazelShTestRule.java;l=40;drc=f7829f855bc31aaf0c0fcef55ef56adec84daa9e) which points to a bundled precompiled `launcher.exe` if Bazel was built for Windows, and to the C++ source code otherwise. This makes a C++ cross-compiler required for a build that is otherwise not dependent on C++. In addition, C++ cross-compilation from Linux to Windows is currently broken (see #19208).

Note that it's possible to run a `genrule` on Windows from a Linux host, showing that this is a testing-specific issue rather than a more general one with cross-platform Windows builds.

A minimal repro is available at https://github.com/jtattermusch/win-from-linux-rbe-repro (thanks @jtattermusch!).

We should consider publishing the `tw.exe`, `xml.exe` and `launcher.exe` binaries to an external repository and fetch it on demand when a test is configured for Windows. This would avoid unnecessarily increasing the Bazel binary size, introducing useless dependencies on other platforms, or requiring a C++ cross compiler only for running tests.

**Comment by jtattermusch** [comment]
Since tw.exe and xml.exe (and launcher.exe) binaries are easily obtained (they are bundled with bazel.exe on windows), it would be good to document a workaround for providing these .exe binaries myself and convincing bazel linux binary to use them in a build.

If lucky, such workaround (if it exists) would allow me to run `sh_test` and 'native_test` rules on windows executor from linux host or it would possibly reveal whether other non-yet-known issues exist with the cross-platform execution.

**Comment by meisterT** [comment]
cc @meteorcloudy

**Comment by meteorcloudy** [comment]
Yeah, I agree we should move stuff out of `@bazel_tools` as much as possible, especially platform specific dependencies, hopefully that is easier in the Bzlmod world.

**Comment by bluecmd** [comment]
+1, I managed to work around this by hacking in tw.exe and xml.exe from the Windows bundle into the installation base on Linux, but that's not really sustainable.

I'd be very grateful if we could fix this once and for all! 😄

**Comment by bluecmd** [comment]
Ping @meteorcloudy - would you welcome a PR for this?

**Comment by meteorcloudy** [comment]
I think we probably want to review a concrete design first for this problem.
